### PR TITLE
RFC84: Data Entries Removal

### PR DESCRIPTION
**Pull request by forus**
In the current implementation of cBioPortal, the functionality for data management is limited to the deletion of entire studies, as documented [here](https://docs.cbioportal.org/data-loading-maintaining-studies/#deleting-a-study). This approach lacks granularity and does not provide users with the ability to selectively remove individual entries associated with a study, such as patients, samples, or specific molecular data profiles. This limitation restricts the flexibility needed for detailed data management and maintenance, leading to potential inefficiencies and challenges in study data handling.

To address this limitation, this PR extends the capabilities of the `cbioportalImporter.py` script and includes additional commands to the java tool that allow for the selective removal of patient or sample.

See more details in the [RFC84  Data Entries Removal document](https://docs.google.com/document/d/1bOIeLJ44r9h3m6dB390kQ_SAAn69KKxYwFTYBTy5-cI/edit).

Reminder: Do not forget to enable the following PR https://github.com/cBioPortal/cbioportal/pull/10820

### CODE REVIEW

File: src/main/java/org/mskcc/cbio/portal/dao/DaoGeneticProfileSamples.java
Comment (pieterlukasse) [inline src/main/java/org/mskcc/cbio/portal/dao/DaoGeneticProfileSamples.java:1]: general question: should there be a warning, or maybe an error if the study happens to contain unsupported data types (ones that might need recalculation)?

Comment (forus) [reply src/main/java/org/mskcc/cbio/portal/dao/DaoGeneticProfileSamples.java:1]: throwing an error sounds like a good idea. Let me implement that.

File: src/main/java/org/mskcc/cbio/portal/dao/DaoPatient.java [hunk 1]
```diff
@@ -215,4 +220,54 @@ private static Patient extractPatient(ResultSet rs) throws SQLException
...
 		}
     }
+
+    public static void removePatientsEverywhereInStudy(int internalStudyId, Set<String> patientStableIds) throws DaoException {
+        log.info("Removing {} patients from study with internal id={} ...", patientStableIds, internalStudyId);
+        Set<Integer> internalPatientIds = findInternalPatientIdsInStudy(internalStudyId, patientStableIds);
+        for (Integer internalPatientId : internalPatientIds) {
+            Set<Integer> patientInternalSampleIds = DaoSample.getSamplesByPatientId(internalPatientId).stream()
+                    .map(Sample::getInternalPatientId).collect(Collectors.toSet());
+            DaoGeneticAlteration.getInstance().removeSamplesInGeneticAlterationsForStudy(internalStudyId, patientInternalSampleIds);
```
Comment (pieterlukasse) [inline src/main/java/org/mskcc/cbio/portal/dao/DaoPatient.java:230]: I think `removeSamplesInGeneticAlterationsForStudy` should be logic belonging to deleteSamples and deletePatients should just call deleteSamples.

**Review by pieterlukasse — COMMENTED** [review]
thanks @forus ! I added some comments. Please take a look

File: src/main/java/org/mskcc/cbio/portal/dao/DaoGeneticProfileSamples.java
Comment (forus) [reply src/main/java/org/mskcc/cbio/portal/dao/DaoGeneticProfileSamples.java:1]: Done https://github.com/cBioPortal/cbioportal-core/pull/46/commits/907de59c0da51046f493e5709705d39beb2cacce

File: src/main/java/org/mskcc/cbio/portal/dao/DaoGeneticProfileSamples.java [hunk 2]
```diff
@@ -126,7 +126,10 @@ public static ArrayList <Integer> getOrderedSampleList(int geneticProfileId) thr
                 String orderedSampleList = rs.getString("ORDERED_SAMPLE_LIST");
 
                 //  Split, based on DELIM token
-                String parts[] = orderedSampleList.split(DELIM);
+                String[] parts = orderedSampleList.split(DELIM);
+                if (parts.length == 1 && parts[0].isBlank()) {
+                    return new ArrayList<Integer>();
```
[the remark this answers is not in the document]
Comment (forus) [reply src/main/java/org/mskcc/cbio/portal/dao/DaoGeneticProfileSamples.java]: Done https://github.com/cBioPortal/cbioportal-core/pull/46/commits/5753d1d1dbab10460aa77a09dec51c25f69b9824

File: src/main/java/org/mskcc/cbio/portal/dao/DaoPatient.java (hunk 1 above)
Comment (forus) [reply src/main/java/org/mskcc/cbio/portal/dao/DaoPatient.java]: Done https://github.com/cBioPortal/cbioportal-core/pull/46/commits/bd54ed240186925755cfcaa16e50ee4a255a1d21

**Review by pieterlukasse — APPROVED** [review]

File: src/main/java/org/mskcc/cbio/portal/dao/DaoGeneticAlteration.java [hunk 3]
```diff
@@ -32,25 +32,39 @@
...
+import java.util.stream.Collectors;
 
 /**
  * Data Access Object for the Genetic Alteration Table.
  *
  * @author Ethan Cerami.
  */
 public class DaoGeneticAlteration {
+
+    private static final Logger log = LoggerFactory.getLogger(DaoGeneticAlteration.class);
```
Comment (pieterlukasse) [inline src/main/java/org/mskcc/cbio/portal/dao/DaoGeneticAlteration.java:66]: logger not used. Did some code get lost by mistake?

File: src/test/java/org/mskcc/cbio/portal/integrationTest/scripts/TestImportCnaDiscreteLongData.java [hunk 4]
```diff
@@ -326,7 +319,6 @@ public void testImportCnaDiscreteLongDataAddsGeneticAlterationsFromNonCnaEvents(
     @Test
     public void testImportCnaDiscreteLongDataIgnoresLineWithDuplicateGene() throws Exception {
         List<TestGeneticAlteration> beforeGeneticAlterations = getAllGeneticAlterations();
-        assertEquals(57, beforeGeneticAlterations.size());
```
Comment (pieterlukasse) [inline src/test/java/org/mskcc/cbio/portal/integrationTest/scripts/TestImportCnaDiscreteLongData.java:329]: should the `beforeGeneticAlterations` variable also be removed? I think so...it is not used anymore. Or you could use it at the end to verify that `afterGeneticAlterations == beforeGeneticAlterations+1` ?

File: src/main/java/org/mskcc/cbio/portal/dao/DaoGeneticAlteration.java (hunk 3 above)
Comment (forus) [reply src/main/java/org/mskcc/cbio/portal/dao/DaoGeneticAlteration.java]: Done https://github.com/cBioPortal/cbioportal-core/pull/46/commits/989604a63beeeca9ec030d4ef21f9e09428763c7

File: src/test/java/org/mskcc/cbio/portal/integrationTest/scripts/TestImportCnaDiscreteLongData.java (hunk 4 above)
Comment (forus) [reply src/test/java/org/mskcc/cbio/portal/integrationTest/scripts/TestImportCnaDiscreteLongData.java:329]: Done https://github.com/cBioPortal/cbioportal-core/pull/46/commits/90aba7e9f33e7e6ea7d3f59bb7e6356329192599

**Review by pieterlukasse — APPROVED** [review]

**Pull request merged**